appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/customview/AppsTransferList.kt

```diff
@@ -0,0 +1,161 @@
+        //TODO check sort
+        apps = apps.sortedByDescending { context.isAppInstalled(it.identifier!!) }
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent?.context)
+                .inflate(R.layout.app_transfer_item, parent, false)
+        return ViewHolder(context, view)
+    }
+
+    override fun getItemCount(): Int {
+        return apps.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
+        val app: EcosystemApp = apps[position]
+        holder?.bind(app)
+        holder?.actionText?.setOnClickListener {
+            presenter.onAppClicked(app)
+        }
+    }
+
+    class ViewHolder(private val context: Context, val view: View) : RecyclerView.ViewHolder(view) {
+        private val icon: ImageView = view.findViewById(R.id.appIcon)
+        private val appName: TextView = view.findViewById(R.id.appName)
+        private val status: TextView = view.findViewById(R.id.status)
+        val actionText: TextView = view.findViewById(R.id.actionText)
+
+        fun bind(app: EcosystemApp) {
+            appName.text = app.name
+            icon.load(app.iconUrl)
+            app.identifier?.let {
+                if (context.isAppInstalled(it)) {
+                    if (app.canTransferKin) {
+                        actionText.background = ContextCompat.getDrawable(context, R.drawable.kin_button_rounded_drawable)
+                        actionText.text = context.getString(R.string.transfer_kin)
+                        TextViewCompat.setTextAppearance(actionText, R.style.kinTextButtonRounded)
+                        actionText.setTextColor(ContextCompat.getColor(context, R.color.kin_btn_text))
+                        actionText.visibility = View.VISIBLE
+                        status.visibility = View.GONE
+                        appName.gravity = Gravity.CENTER_VERTICAL
+                    } else {
+                        appName.gravity = Gravity.TOP
+                        status.visibility = View.VISIBLE
+                        status.text = context.getString(R.string.coming_soon)
+                        actionText.visibility = View.GONE
+                    }
+                } else {
+                    appName.gravity = Gravity.TOP
+                    TextViewCompat.setTextAppearance(actionText, R.style.kinTextButtonRounded_Hollow)
+                    actionText.background = ContextCompat.getDrawable(context, R.drawable.kin_button_rounded_hollow_drawable)
+                    actionText.text = context.getString(R.string.learn_more)
+                    status.text = context.getString(R.string.not_installed)
+                    actionText.setTextColor(ContextCompat.getColor(context, R.color.kin_btn_hollow_text))
+                    status.visibility = View.VISIBLE
+                    actionText.visibility = View.VISIBLE
+                }
```

Comment:
suggest that you extract the contents of the above three if/else cases to methods